client/payment-details/transaction-breakdown/index.tsx

```diff
@@ -0,0 +1,319 @@
+	}
+
+	if (
+		undefined === captureEvent ||
+		undefined === captureEvent.transaction_details ||
+		undefined === captureEvent.fee_rates
+	) {
+		return <div />;
+	}
+
+	const formattedStoreAmount =
+		formatCurrency(
+			captureEvent.transaction_details.store_amount,
+			captureEvent.transaction_details.store_currency
+		) +
+		' ' +
+		captureEvent.transaction_details.store_currency;
+
+	const formattedCustomerAmount =
+		formatCurrency(
+			captureEvent.transaction_details.customer_amount,
+			captureEvent.transaction_details.customer_currency,
+			captureEvent.transaction_details.store_currency
+		) +
+		' ' +
+		captureEvent.transaction_details.customer_currency;
+
+	const isMultiCurrency =
+		captureEvent.transaction_details.store_currency !==
+		captureEvent.transaction_details.customer_currency;
+
+	const formattedAmount =
+		formattedCustomerAmount +
+		( isMultiCurrency ? ` → ${ formattedStoreAmount }` : '' );
+
+	const feeExchangeRate = captureEvent.fee_rates.fee_exchange_rate?.rate || 1;
+
+	const conversionRate = isMultiCurrency ? (
+		<FlexItem className="wcpay-transaction-breakdown__conversion_rate">
+			{ ' @ 1 ' }
+			{ captureEvent.transaction_details.customer_currency }
+			{ ' → ' }
+			{ Math.round( ( 1 / feeExchangeRate ) * 1000000 ) / 1000000 }
+			{ '	' }
+			{ captureEvent.transaction_details.store_currency }
+		</FlexItem>
+	) : (
+		''
+	);
+
+	function formatFeeType(
+		type: string,
+		additionalType: string | undefined
+	): string {
+		if ( 'total' === type ) {
+			return __( 'Total transaction fee', 'woocommerce-payments' );
+		}
+		if ( 'base' === type ) {
+			return __( 'Base fee', 'woocommerce-payments' );
+		}
+		if ( 'additional' === type && 'international' === additionalType ) {
+			return __( 'International card fee', 'woocommerce-payments' );
+		}
+		if ( 'additional' === type && 'fx' === additionalType ) {
+			return __( 'Currency conversion fee', 'woocommerce-payments' );
+		}
+		return __( 'Fee', 'woocommerce-payments' );
+	}
+
+	function formatFeeRate(
+		percentage: number,
+		fixed: number,
+		currency: string,
+		storeCurrency: string
+	): string {
+		const formattedPercentage = percentage
+			? Math.round( percentage * 10000 ) / 100 + '%'
+			: '';
+		const formattedFixed = fixed
+			? formatCurrency( fixed, currency, storeCurrency )
+			: '';
+		return join(
+			filter( [ formattedPercentage, formattedFixed ], Boolean ),
+			' + '
+		);
+	}
+
+	function formatFee(
+		type: string,
+		additionalType: string | undefined,
+		percentage: number,
+		fixed: number,
+		currency: string,
+		storeCurrency: string,
+		amount?: number
+	): JSX.Element[] {
+		const formattedFeeType = formatFeeType( type, additionalType );
+		const formattedFeeRate = formatFeeRate(
+			percentage,
+			fixed,
+			currency,
+			storeCurrency
+		);
+		const formattedFeeAmount = amount
+			? ' - ' + formatCurrency( amount, storeCurrency, storeCurrency )
+			: '';
+
+		return [
+			<Flex
+				key="{ type }_fee_info"
```

Comment:
In the current form this is a string literal, but I think the intent is to generate a key based on the value of `type`. So it should be:

```suggestion
				key={`${type}_fee_info`}
```

This appears to be the case for the generation of other keys below

Reply:
Thanks! But the whole function was changed into a different style in 64c70305b, removing the need for the key.